Rock/Reporting/DataSelect/GroupMember/GroupLinkSelect.cs

```diff
@@ -0,0 +1,126 @@
+    [Export( typeof( DataSelectComponent ) )]
+    [ExportMetadata( "ComponentName", "Select Group Name" )]
+    [BooleanField( "Show As Link", "", true )]
+    public class GroupLinkSelect : DataSelectComponent
+    {
+        /// <summary>
+        /// Gets the name of the entity type. Filter should be an empty string
+        /// if it applies to all entities
+        /// </summary>
+        /// <value>
+        /// The name of the entity type.
+        /// </value>
+        public override string AppliesToEntityType
+        {
+            get
+            {
+                return typeof( Rock.Model.GroupMember ).FullName;
+            }
+        }
+
+        /// <summary>
+        /// The PropertyName of the property in the anonymous class returned by the SelectExpression
+        /// </summary>
+        /// <value>
+        /// The name of the column property.
+        /// </value>
+        public override string ColumnPropertyName
+        {
+            get
+            {
+                return "Group Name";
+            }
+        }
+
+        /// <summary>
+        /// Gets the section that this will appear in in the Field Selector
+        /// </summary>
+        /// <value>
+        /// The section.
+        /// </value>
+        public override string Section
+        {
+            get
+            {
+                return "Common";
+            }
+        }
+
+        /// <summary>
+        /// Gets the grid field.
+        /// </summary>
+        /// <param name="entityType">Type of the entity.</param>
+        /// <param name="selection">The selection.</param>
+        /// <returns></returns>
+        public override System.Web.UI.WebControls.DataControlField GetGridField( Type entityType, string selection )
+        {
+            var result = new BoundField();
+            
+            // Disable encoding of field content because the value contains markup.
+            result.HtmlEncode = false;
+            
+            return result;
+        }
+
+        /// <summary>
+        /// Gets the expression.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="entityIdProperty">The entity identifier property.</param>
+        /// <param name="selection">The selection.</param>
+        /// <returns></returns>
+        public override System.Linq.Expressions.Expression GetExpression( Data.RockContext context, System.Linq.Expressions.MemberExpression entityIdProperty, string selection )
+        {
+            bool showAsLink = this.GetAttributeValueFromSelection( "ShowAsLink", selection ).AsBooleanOrNull() ?? false;
+
+            var memberQuery = new GroupMemberService( context ).Queryable();
+
+            IQueryable<string> groupLinkQuery;
+
+            if ( showAsLink )
+            {
+                // Return a string in the format: <a href='/group/{groupId}'>Group Name</a>
+                // Prepend with a HTML comment to ensure that items are sorted correctly.
```

Comment:
@MrUpsideDown I suspected you might be making this for v3.  However, since you are doing a pull request into v4, you can safely make the change now, but feel free to make the call on that.